argilla-server/pyproject.toml

```
@@ -22,7 +22,8 @@ maintainers = [{ name = "argilla", email = "contact@argilla.io" }]
 dependencies = [
     # Basic dependencies
     "fastapi ~= 0.115.0",
-    "pydantic ~= 1.10.18",
+    "pydantic ~= 2.9.0",
+    "pydantic-settings ~= 2.6.0",
```

Something similar for `pydantic-settings`:

```suggestion
    "pydantic-settings ~= 2.6.1",
```

Don't forget to do `pdm install` after change this, so the lock file is updated.